sjsonnet/src/sjsonnet/Evaluator.scala

```diff
@@ -314,73 +337,120 @@ class Evaluator(parseCache: collection.mutable.HashMap[(Path, String), fastparse
       )
   }
 
-  def visitBinaryOp(pos: Position, lhs: Expr, op: BinaryOp.Op, rhs: Expr)(implicit scope: ValScope) = {
-    (visitExpr(lhs), op, visitExpr(rhs)) match {
+  def visitBinaryOp(pos: Position, lhs: Expr, op: Int, rhs: Expr)(implicit scope: ValScope) = {
+    val l = visitExpr(lhs)
+    val r = visitExpr(rhs)
+    def fail() = Error.fail(s"Unknown binary operation: ${l.prettyName} ${Expr.BinaryOp.name(op)} ${r.prettyName}", pos)
+    op match {
```

Comment:
Should this be annotated `: @switch` to ensure the `tableswitch`-based compilation isn't accidentally broken?

Reply:
AFAIR `@switch` is broken. But even if it works, it also accepts a `lookupswitch` so it's not much of help.